scripts/util/simulate_expression.R

```diff
@@ -4,264 +4,206 @@
-    } else {
-      group_params <- sampleGroupMatrix(n, mean_df, sd_df)
-      group_df <- group_params[[1]]
-      group_info <- group_params[[2]]
-    } 
-  } else {
-      group_df <- c()
-      group_info <- c()
-  }
-
-  # Get Random Noise Features
-  rand_df <- c()
-  if (r > 0) {
-    for (rand_idx in 1:r) {
-      rand_vector <- runif(n, min = 0, max = 1)
-      rand_df <- cbind(rand_df, rand_vector)
-    }
-    colnames(rand_df) <- paste('random', seq(1, ncol(rand_df)), sep = "_")
-    rand_df <- tibble::as_data_frame(rand_df)
+  if (num_remainder > 0) {
+    sample_labels <- sort(c(sample_labels, LETTERS[1:num_remainder]))
   }
 
-  # Get Continuous Function Features
-  cont_df <- c()
-  cont_other_df <- c()
-  if (length(func_list) > 0) {
-    for (cont_idx in 1:length(func_list)) {
-      continuous_rand_x <- runif(n, min = -1, max = 1)
-      continuous_rand_y <- func_list[[cont_idx]](continuous_rand_x)
-      
-      cont_df <- cbind(cont_df, continuous_rand_y)
-      cont_other_df <- cbind(cont_other_df, continuous_rand_x)
-    }
-    colnames(cont_df) <- paste('continuous', seq(1, ncol(cont_df)), sep = "_")
-    cont_df <- tibble::as_data_frame(cont_df)
-    
-    colnames(cont_other_df) <- paste('continuous_domain',
-                                     seq(1, ncol(cont_other_df)), sep = "_")
-    cont_other_df <- tibble::as_data_frame(cont_other_df)
-  }
+  gene_labels <- x_1$allLabels
 
-  # Get Presence/Absence of a Features
-  pres_df <- c()
-  if (b > 0) {
-    for (pres_idx in 1:b) {
-      rand_presence <- rnorm(n, mean = 3, sd = 0.5)
-      rand_zeroone <- sample(c(0, 1), n, replace = TRUE)
-      
-      rand_presence <- rand_presence * rand_zeroone
-      pres_df <- cbind(pres_df, rand_presence)
-    }
-    colnames(pres_df) <- paste('presence', seq(1, ncol(pres_df)), sep = "_")
-    pres_df <- tibble::as_data_frame(pres_df)
+  # Combine Matrix
+  x_matrix <- tibble::as_data_frame(rbind(x_1$datExpr,
+                                          x_2$datExpr))
+  colnames(x_matrix) <- gsub('[.]', '_', colnames(x_matrix))
+  x_matrix$groups <- sample_labels
+  x_matrix <- rbind(gene_labels, x_matrix)
+  
+  if (return_eigen) {
+    return(list(x_matrix, eigen_gene_samples))
   }
+  
+  return(x_matrix)
+}
 
-  # Get cell-type Features
-  if (sum(cell_type_mean_df + cell_type_sd_df) != 0) {
-    if (all(dim(cell_type_mean_df) != dim(cell_type_sd_df))) {
-      stop("provide the same cell-type parameter dimensions")
-    } else {
-      # This will generate cell-types and then automatically simulate
-      # differential cell-type proportion
-      cell_type_info <- sampleCellMatrix(n, cell_type_mean_df, cell_type_sd_df)
-      cell_type_df <- cell_type_info[[1]]
-      cell_type_proportion <- cell_type_info[[2]]
+# Vary sample size and amount of background noise
+ns <- c(250, 500, 1000, 2000, 4000)
+background_noises <- c(0, 0.1, 0.5, 1, 3)
+genes <- c(500, 1000)
+
+# Other constants
```

Comment:
It might be a little "cleaner" to set these as defaults for `simulateExpression` but up to you

Reply:
Yeah, it could be cleaner in the sense of less lines, but I think separating out the parameters from the function makes it a bit easier to read/understand.